client/src/lib/api/files.ts

```diff
@@ -32,6 +43,36 @@ export async function getSignedUploadURLs(user: User, fileNames: String[]) {
   return new Map<string, string>(Object.entries(result));
 }
 
+/**
+ * Get all files from the given dataset for the given user
+ *
+ * @param user User whose files need to be retrieved
+ * @returns A Promise that resolves to a list of UserFile objects
+ */
+export async function getFilesFromDataset(
+  user: User,
+  datasetName: string
+): Promise<UserFile[]> {
+  const getDataset = async () => {
+    const docRef = doc(firestore, `users/${user!.uid}/datasets/${datasetName}`);
+    const docSnapShot = await getDoc(docRef);
+    const returnedDocument: DocumentData = docSnapShot.data() as DocumentData;
+    return returnedDocument as Dataset;
+  };
+  console.log(user);
+  console.log(datasetName);
+  const dataset: Dataset = await getDataset();
+  console.log(dataset);
+
+  const returnedFiles = new Array<UserFile>();
+  for (const filename of dataset.files) {
```

Comment:
Prefer using `map` where possible to an imperative solution as given here. So in this case, we call map on the filenames, converting them into an array of Files, and we can return that directly without having to give that variable a name: e.g.

```
return dataset.files.map((file) => getFile(file.name)) as UserFile[]